core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala

```diff
@@ -427,12 +429,17 @@ class CleanerTester(
 
     def broadcastCleaned(broadcastId: Long): Unit = {
       toBeCleanedBroadcstIds -= broadcastId
-      logInfo("Broadcast" + broadcastId + " cleaned")
+      logInfo("Broadcast " + broadcastId + " cleaned")
     }
 
     def accumCleaned(accId: Long): Unit = {
       logInfo("Cleaned accId " + accId + " cleaned")
     }
+
+    def checkpointCleaned(rddId: Long): Unit = {
+      toBeCheckpointIds -= rddId
```

Comment:
@srowen  When the checkpoint is cleaned, `checkpointCleaned ` is called asynchronous,  `isAllCleanedUp` returns true,  and then `postGCTester.assertCleanup()`  return.